porcupine/plugins/filemanager.py

```diff
@@ -70,6 +83,24 @@ def select_name() -> None:
     ok_button = ttk.Button(button_frame, text="OK", command=select_name, state="disabled", width=1)
     ok_button.pack(side="right", expand=True, fill="x", padx=(5, 0))
 
+    def overwriting() -> None:
+        entry.config(state="disabled")
+        ok_button.config(state="enabled")
+
+    def renaming() -> None:
+        entry.config(state="enabled")
+        validate_name()
```

Comment:
Maybe `validate_name()` should run every time the `choice` variable changes? You can do that with `choice.trace_add`, just like how there is already `var.trace_add`.

Feel free to rename `var`. It is no longer clear what it is because there are two tkinter variables.

Reply:
Not always when it changes, but only when passing from `overwrite_var = False` to `True` (this is the new name of the variable).

Reply:
Why not always? Why should it update when switching from `True` to `False`, but stay as is if you then switch back to `True`?

Reply:
Here's how I imagined this:
- `overwriting()` and `renaming()` set the entry to enabled or disabled.
- `validate_name()` should sets `ok_button` to enabled or disabled. Feel free to rename it to `update_ok_button_state()` or something.

This way it would be very clear what each function does, because the state of `ok_button` would be updated from one function only. Then it makes sense to run `validate_name()` whenever any of the two variables change, simply because values of both variables are used when deciding the state of the button.

Reply:
So `overwrite_var == True` if the overwrite `RadioButton` is checked. When it's checked I thought that the `OK` button to go on should always be enabled since it's always possible to overwrite. Let's say that I want to copy `file.txt` where there's another file with this name. I have overwrite checked as default. Then I check 'change name of destination' and modify the new name to `filex.txt`. OK is enabled. Then i modify back to `file.txt` and `validate_name()` disables `OK`. Now `overwrite_var == False` since I'm modifying the name. When i click again the `RadioButton` for overwriting I want the `OK` button to be enabled no matter what's written in the entry field (also the entry field is disabled) so if I call again `validate_name()` it will disable `OK`.

Reply:
You wrote while I was writing :) yes, I can refactor it like this.